// cheese-core/js/core/keys.js
export default class keys {
    constructor() {
    }
    static homeAsync() {
        return KeysApi.homeAsync();
    }
    static home() {
        return KeysApi.home();
    }
    static backAsync() {
        return KeysApi.backAsync();
    }
    static back() {
        return KeysApi.back();
    }
    static quickSettingsAsync() {
        return KeysApi.quickSettingsAsync();
    }
    static quickSettings() {
        return KeysApi.quickSettings();
    }
    static powerDialogAsync() {
        return KeysApi.powerDialogAsync();
    }
    static powerDialog() {
        return KeysApi.powerDialog();
    }
    static pullNotificationBarAsync() {
        return KeysApi.pullNotificationBarAsync();
    }
    static pullNotificationBar() {
        return KeysApi.pullNotificationBar();
    }
    static recentsAsync() {
        return KeysApi.recentsAsync();
    }
    static recents() {
        return KeysApi.recents();
    }
    static lockScreenAsync() {
        return KeysApi.lockScreenAsync();
    }
    static lockScreen() {
        return KeysApi.lockScreen();
    }
    static screenShotAsync() {
        return KeysApi.screenShotAsync();
    }
    static screenShot() {
        return KeysApi.screenShot();
    }
    static splitScreenAsync() {
        return KeysApi.splitScreenAsync();
    }
    static splitScreen() {
        return KeysApi.splitScreen();
    }
}
