// cheese-core/ts/core/class.ts
declare var ClassApi: any
import {Class, List} from "types";
declare type ClassUtils = {
    clz: string;
    new: (...args:any[]) => any;
    set:(nodeName: string,name?: string)=>string
}
export default class class_ {
    constructor() {
    }

    public static readonly Boolean:Class<boolean> = ClassApi._Boolean
    public static readonly String:Class<string> = ClassApi._String
    public static readonly Float:Class<number> = ClassApi._Float
    public static readonly Long:Class<number> = ClassApi._Long
    public static readonly ByteArray:Class<any> = ClassApi._ByteArray

    public static getClassListAsync(packageName: string): Promise<List<string>> {
        return ClassApi.getClassListAsync(packageName);
    }

    public static getClassList(packageName: string): List<string> {
        return ClassApi.getClassList(packageName);
    }

    public static loadDexAsync(path: string): Promise<void> {
        return ClassApi.loadDexAsync(path);
    }

    public static loadDex(path: string): void {
        return ClassApi.loadDex(path);
    }

    public static hasClassAsync(className: string): Promise<boolean> {
        return ClassApi.hasClass(className);
    }

    public static hasClass(className: string): boolean {
        return ClassApi.hasClass(className);
    }

    public static findClassAsync(className: string): Promise<ClassUtils> {
        return ClassApi.findClassAsync(className);
    }

    public static findClass(className: string): ClassUtils {
        return ClassApi.findClass(className);
    }


}
