// cheese-core/ts/core/termux.ts
import {Callback} from "types";

declare var TermuxApi: any
export default class termux {
    private that: any;
    constructor() {
        this.that=TermuxApi.new()
    }

    public  connectTermux(callback: Callback): void {
        return this.that.connectTermux(callback);
    }

    public  close(): void {
        return this.that.close();
    }



    public  state(): boolean {
        return this.that.state();
    }


    public  read(): string {
        return this.that.read();
    }

    public  send(c: string): void {
        return this.that.send(c);
    }

    public  whileRead(callback: Callback): void {
        return this.that.whileRead(callback);
    }

    public  stopWhileRead(): void {
        return this.that.stopWhileRead();
    }
    public  whileReadState(): boolean {
        return this.that.whileReadState();
    }

    public  install(pkg: string): void {
        return this.that.install(pkg);
    }

    public  uninstall(pkg: string): void {
        return this.that.uninstall(pkg);
    }


}
