// cheese-core/js/app.js
export default class app {
    constructor() {
    }
    static openUrlAsync(url) {
        return AppApi.openUrlAsync(url);
    }
    static openUrl(url) {
        return AppApi.openUrl(url);
    }
    static uninstallAsync(pkg) {
        return AppApi.uninstallAsync(pkg);
    }
    static uninstall(pkg) {
        return AppApi.uninstall(pkg);
    }
    static getPackageNameAsync(appName) {
        return AppApi.getPackageNameAsync(appName);
    }
    static getPackageName(appName) {
        return AppApi.getPackageName(appName);
    }
    static getAppNameAsync(packageName) {
        return AppApi.getAppNameAsync(packageName);
    }
    static getAppName(packageName) {
        return AppApi.getAppName(packageName);
    }
    static openAppSettingsAsync(packageName) {
        return AppApi.openAppSettingsAsync(packageName);
    }
    static openAppSettings(packageName) {
        return AppApi.openAppSettings(packageName);
    }
    static openAppAsync(pkg) {
        return AppApi.openAppAsync(pkg);
    }
    static openApp(pkg) {
        return AppApi.openApp(pkg);
    }
    static openSchemeAsync(schemeUri) {
        return AppApi.openSchemeAsync(schemeUri);
    }
    static openScheme(schemeUri) {
        return AppApi.openScheme(schemeUri);
    }
}
