// cheese-core/js/converters.js
export default class converters {
    constructor() {
    }
    static pairArrayAsync(...data) {
        return ConvertersApi.pairArrayAsync(...data);
    }
    static pairArray(...data) {
        return ConvertersApi.pairArray(...data);
    }
    static pairArraysAsync(...data) {
        return ConvertersApi.pairArraysAsync(...data);
    }
    static pairArrays(...data) {
        return ConvertersApi.pairArrays(...data);
    }
    static sdToStreamAsync(filePath) {
        return ConvertersApi.sdToStreamAsync(filePath);
    }
    static sdToStream(filePath) {
        return ConvertersApi.sdToStream(filePath);
    }
    static assetsToStreamAsync(filePath) {
        return ConvertersApi.assetsToStreamAsync(filePath);
    }
    static assetsToStream(filePath) {
        return ConvertersApi.assetsToStream(filePath);
    }
    static streamToBitmapAsync(inputStream) {
        return ConvertersApi.streamToBitmapAsync(inputStream);
    }
    static streamToBitmap(inputStream) {
        return ConvertersApi.streamToBitmap(inputStream);
    }
    static bitmapToStreamAsync(bitmap) {
        return ConvertersApi.bitmapToStreamAsync(bitmap);
    }
    static bitmapToStream(bitmap) {
        return ConvertersApi.bitmapToStream(bitmap);
    }
}
