// cheese-core/js/point.js
export default class point {
    constructor() {
    }
    static swipeToPointAsync(sx, sy, ex, ey, dur) {
        return PointApi.swipeToPointAsync(sx, sy, ex, ey, dur);
    }
    static swipeToPoint(sx, sy, ex, ey, dur) {
        return PointApi.swipeToPoint(sx, sy, ex, ey, dur);
    }
    static clickPointAsync(x, y) {
        return PointApi.clickPointAsync(x, y);
    }
    static clickPoint(x, y) {
        return PointApi.clickPoint(x, y);
    }
    static longClickPointAsync(x, y) {
        return PointApi.longClickPointAsync(x, y);
    }
    static longClickPoint(x, y) {
        return PointApi.longClickPoint(x, y);
    }
    static touchDownAsync(x, y) {
        return PointApi.touchDown(x, y);
    }
    static touchDown(x, y) {
        return PointApi.touchDown(x, y);
    }
    static touchMoveAsync(x, y) {
        return PointApi.touchMove(x, y);
    }
    static touchMove(x, y) {
        return PointApi.touchMove(x, y);
    }
    static touchUpAsync() {
        return PointApi.touchUpAsync();
    }
    static touchUp() {
        return PointApi.touchUp();
    }
}
