// cheese-core/js/device.js
export default class device {
    constructor() {
    }
    static getIMEIAsync() {
        return DeviceApi.getIMEIAsync();
    }
    static getIMEI() {
        return DeviceApi.getIMEI();
    }
    static supportedOAIDAsync() {
        return DeviceApi.supportedOAIDAsync();
    }
    static supportedOAID() {
        return DeviceApi.supportedOAID();
    }
    static getOAIDAsync() {
        return DeviceApi.getOAIDAsync();
    }
    static getOAID() {
        return DeviceApi.getOAID();
    }
    static getPositionAsync() {
        return DeviceApi.getPositionAsync();
    }
    static getPosition() {
        return DeviceApi.getPosition();
    }
    static getPublicIPAsync(url = "https://api.ipify.org") {
        return DeviceApi.getPublicIPAsync(url);
    }
    static getPublicIP(url = "https://api.ipify.org") {
        return DeviceApi.getPublicIP(url);
    }
    static getWifiIPAsync() {
        return DeviceApi.getWifiIPAsync();
    }
    static getWifiIP() {
        return DeviceApi.getWifiIP();
    }
    static getAndroidVersionAsync() {
        return DeviceApi.getAndroidVersionAsync();
    }
    static getAndroidVersion() {
        return DeviceApi.getAndroidVersion();
    }
    static getStatusBarHeightAsync() {
        return DeviceApi.getStatusBarHeightAsync();
    }
    static getStatusBarHeight() {
        return DeviceApi.getStatusBarHeight();
    }
    static getNavigationBarHeightAsync() {
        return DeviceApi.getNavigationBarHeightAsync();
    }
    static getNavigationBarHeight() {
        return DeviceApi.getNavigationBarHeight();
    }
    static getScreenHeightAsync() {
        return DeviceApi.getScreenHeightAsync();
    }
    static getScreenHeight() {
        return DeviceApi.getScreenHeight();
    }
    static getScreenWidthAsync() {
        return DeviceApi.getScreenWidthAsync();
    }
    static getScreenWidth() {
        return DeviceApi.getScreenWidth();
    }
    static getScreenDpiAsync() {
        return DeviceApi.getScreenDpiAsync();
    }
    static getScreenDpi() {
        return DeviceApi.getScreenDpi();
    }
    static getTimeAsync() {
        return DeviceApi.getTimeAsync();
    }
    static getTime() {
        return DeviceApi.getTime();
    }
    static getClipboardAsync() {
        return DeviceApi.getClipboardAsync();
    }
    static getClipboard() {
        return DeviceApi.getClipboard();
    }
    static setClipboardAsync(text) {
        return DeviceApi.setClipboardAsync(text);
    }
    static setClipboard(text) {
        return DeviceApi.setClipboard(text);
    }
}
