// cheese-core/js/files.js
export default class files {
    constructor() {
    }
    static readAsync(path) {
        return FilesApi.readAsync(path);
    }
    static read(path) {
        return FilesApi.read(path);
    }
    static rmAsync(path) {
        return FilesApi.rm(path);
    }
    static rm(path) {
        return FilesApi.rm(path);
    }
    static createAsync(path) {
        return FilesApi.createAsync(path);
    }
    static create(path) {
        return FilesApi.create(path);
    }
    static copyAsync(sourceDirPath, destinationDirPath) {
        return FilesApi.copyAsync(sourceDirPath, destinationDirPath);
    }
    static copy(sourceDirPath, destinationDirPath) {
        return FilesApi.copy(sourceDirPath, destinationDirPath);
    }
    static readJsonAsync(filePath, keys) {
        return FilesApi.readJsonAsync(filePath, keys);
    }
    static readJson(filePath, keys) {
        return FilesApi.readJson(filePath, keys);
    }
    static isFileAsync(path) {
        return FilesApi.isFileAsync(path);
    }
    static isFile(path) {
        return FilesApi.isFile(path);
    }
    static isFolderAsync(path) {
        return FilesApi.isFolderAsync(path);
    }
    static isFolder(path) {
        return FilesApi.isFolder(path);
    }
    static appendAsync(filePath, content) {
        return FilesApi.append(filePath, content);
    }
    static append(filePath, content) {
        return FilesApi.append(filePath, content);
    }
    static writeAsync(filePath, content) {
        return FilesApi.writeAsync(filePath, content);
    }
    static write(filePath, content) {
        return FilesApi.write(filePath, content);
    }
    static saveAsync(obj, path) {
        return FilesApi.saveAsync(obj, path);
    }
    static save(obj, path) {
        return FilesApi.save(obj, path);
    }
}
